src/app/storage/FabricTableImpl.ipp

```diff
@@ -796,11 +793,25 @@ void FabricTableImpl<StorageId, StorageData, kIteratorsMax>::SetTableSize(uint16
     mMaxPerFabric   = std::min(endpointTableSize, std::min(Serializer::kMaxPerFabric(), maxPerFabric));
 }
 
-template <class StorageId, class StorageData, size_t kIteratorsMax>
-FabricTableImpl<StorageId, StorageData, kIteratorsMax>::EntryIteratorImpl::EntryIteratorImpl(
-    FabricTableImpl & provider, FabricIndex fabricIdx, EndpointId endpoint, uint16_t maxPerFabric, uint16_t maxPerEndpoint) :
+template <class StorageId, class StorageData>
+template <size_t kEntryMaxBytes>
+CHIP_ERROR FabricTableImpl<StorageId, StorageData>::IterateEntries(FabricIndex fabric, PersistentStore<kEntryMaxBytes> & store,
+                                                                   IterateFnType iterateFn)
+{
+    VerifyOrReturnError(IsInitialized(), CHIP_ERROR_INTERNAL);
+
+    EntryIteratorImpl<kEntryMaxBytes> iterator(*this, fabric, mEndpointId, mMaxPerFabric, mMaxPerEndpoint, store);
+    AutoRelease<EntryIteratorImpl<kEntryMaxBytes>> iteratorRelease(&iterator);
```

Comment:
```suggestion
    // Make sure to call Release() on the iterator before deleting it, so that it can clean up any internal
    // allocations it might have made.
    AutoRelease<EntryIteratorImpl<kEntryMaxBytes>> iteratorRelease(&iterator);
```

I'm still not quite following how template specialization would affect this setup here, unless someone specializes EntryIteratorImpl for a particular number of entry bytes?  I feel like I am missing something there.

The other reason I am asking these questions and ending up with some confusion is that the intended (but perhaps insufficiently clearly documented?) semantics of `CommonIterator::Release` are that it calls the object's destructor and frees up the memory (back to the heap or a pool).  But that's not at all what's going on here.